_source/_posts/2022-05-19-angular-microfrontend-deploy.md

```diff
@@ -0,0 +1,292 @@
+---
+layout: blog_post
+title: "Secure and Deploy Micro Frontends with Angular"
+author: alisa-duncan
+by: advocate
+communities: [javascript]
+description: "A tutorial to prepare your micro frontend Angular project for production by using dynamic module loading, with deployment configured using Vercel. "
+tags: [angular, typescript]
+tweets:
+- "Ship it! 🚀  Let's deploy an #Angular micro frontend project using @vercel"
+- "Deploy an @angular project using micro frontends! This walkthrough configures an #Angular micro frontend project for multiple environments and deploys to @vercel. 👀"
+image: blog/angular-microfrontend-deploy/angular-microfrontend-deploy-social.jpg
+type: conversion
+github: https://github.com/oktadev/okta-angular-microfrontend-example
+---
+
+Micro frontends continue to gain interest and traction in front-end development. The architecture models the same concept as micro services - as a way to decompose monolithic front-end applications. And just like with micro services, micro frontends have complexities to manage.
+
+This post is part two in a series about building an e-commerce site with Angular using micro frontends. We use Webpack 5 with Module Federation to wire the micro frontends together, demonstrate sharing authenticated state between the different front ends, prepare for deployment using dynamic module loading, and deploy it all to [Vercel](https://vercel.com/)'s free hosting plan. 
+
+In this second post, we're building on the site we created in part one, [How to Build Micro Frontends Using Module Federation in Angular](/blog/2022/05/02/angular-microfrontend-auth). Let's add dynamic module loading to our cupcake e-commerce site, secure unprotected routes, and set up the site for deployment in Vercel, a service designed to improve the front-end development workflow. 🎉
+
+In the end, you'll have an app that looks like this publicly available through Vercel:
+
+{% img blog/angular-microfrontend-deploy/final-app.gif alt:"Animated gif showing the final project with micro frontends that handle the cupcake shopping basket, authentication, and profile information" width:"800" %}{: .center-image }
+
+**Prerequisites**
+
+- [Node](https://nodejs.org/) This project was developed using Node v16.14 with npm v8.5 
+- [Angular CLI](https://angular.io/cli)
+- [GitHub Account](https://github.com/)
+- [Vercel Account](https://vercel.com/) connected through your GitHub credentials for automating deployment.
+
+{% include toc.md %}
+
+## Review the Angular micro-frontends project using Webpack and Module Federation
+
+Let's start by refreshing our memories—dust off your project from the first post. Just like last time, we'll need both IDE and the terminal. 
```

Comment:
```suggestion
Let's start by refreshing our memories—dust off your project from the [first post](https://developer.okta.com/blog/2022/05/17/angular-microfrontend-auth). Just like last time, we'll need both IDE and the terminal. 
```
does it make sense to link back at first mention...? (for folks who land on the second one first...)